src/main/scala/rise/core/types/Type.scala

```diff
@@ -105,25 +113,9 @@ sealed case class VectorType(size: Nat, elemType: DataType) extends BasicType {
   override def toString: String = s"<$size>$elemType"
 }
 
-object int2 extends VectorType(2, int)
-
-object int3 extends VectorType(3, int)
-
-object int4 extends VectorType(4, int)
-
-object int8 extends VectorType(8, int)
-
-object int16 extends VectorType(16, int)
-
-object float2 extends VectorType(2, float)
-
-object float3 extends VectorType(3, float)
-
-object float4 extends VectorType(4, float)
-
-object float8 extends VectorType(8, float)
-
-object float16 extends VectorType(16, float)
+object vec {
+  def apply(size: Nat, elemType: DataType) = VectorType(size, elemType)
```

Comment:
Do we want to enforce here that `size` is a `Cst` and that it is one of: `2, 3, 4, 8, 16`?

Reply:
Maybe we want to write some vector-width generic code with types such as `nFunT(w => f32 ->: vec(w, f32)`? Although you could still use `VectorType(w, f32)` if you wanted.

Reply:
I feel like such checks would be better suited when you try to generate code with a specific backend/target?

Reply:
This form of polymorphism is dangerous anyway because the type suggests that we can build a `VectorType(w, f32)` for all `w: Nat`. But that isn't true.

We will have to be looking a bit more carefully how to define the `VectorType` on the formal side, I guess.

Reply:
Maybe we can open an issue about that? There is also the issue that `elemType` should be a `ScalarType` ideally but we have no way of creating `ScalarTypeIdentifiers` for example.

Reply:
Maybe we want subtyping constraints (https://github.com/rise-lang/rise/issues/11) but also nat constraints?